### PR TITLE
Adding BlocklyActivityHelper.

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Extracting several Blockly initialization methods out of AbstractBlocklyActivity and into BlocklyActivityHelper, to make it easier to add Blockly to other Activity classes.

---
This change is [Reviewable](https://reviewable.io/reviews/google/blockly-android/511)

### CODE REVIEW

File: blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+     * {@link #mWorkspaceFragment}, and optionally, {@link #mToolboxFlyoutFragment} and
+     * {@link #mTrashFragment}. This base implementation attempts to acquire references to the
+     * {@link #mToolboxFlyoutFragment} and {@link #mTrashFragment} using the layout ids
+     * {@link R.id#} and {@link R.id#blockly_trash}, respectively. Subclasses may leave these
+     * {@code null} if the views are not present in the UI.
+     * <p>
+     * Always called once from {@link #BlocklyActivityHelper} and before {@link #mController} is
+     * instantiated.
+     */
+//    the base implementation of {@link #onCreateFragments()} looks for
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java]: Delete this duplicate comment.

File: blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+    /**
+     * @return True if the action was handled to close a previously open (and closable) flyout.
+     *         Otherwise false.
+     */
+    public boolean onBackToCloseFlyouts() {
+        return mController.closeFlyouts();
+    }
+
+    public void requestCodeGeneration(
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java]: Add javadoc

File: blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java [hunk 3]
```diff
@@ -807,14 +617,6 @@ protected boolean onBackToCloseNavMenu() {
...
 
-    /**
-     * @return True if the action was handled to close a previously open (and closable) flyout.
-     *         Otherwise false.
-     */
-    protected boolean onBackToCloseFlyouts() {
-        return mController.closeFlyouts();
-    }
-
     private class DefaultVariableCallback extends BlocklyController.VariableCallback {
```
Comment (RoboErikG) [inline blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java]: As a TODO we should pull this out into it's own file. Maybe DialogVariableCallback?

File: blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java (hunk 1 above)
Comment (AnmAtAnm) [inline blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java]: Merged two comments, with elaboration.

**Pull request merged**